More gdb introspection on CI

azat opened the pull request:
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

This is required to get more details for things like from:
- #31531
- #33057

```
Failed assertion: "bin->low_bits_full > bin->low_bits_empty" Received signal 6 Received signal Aborted (6)
```

Cc: @tavplubix

azat committed:
test/fuzzer: preserve core

This may help to capture things like in [1]:

    Failed assertion: "bin->low_bits_full > bin->low_bits_empty" Received signal 6 Received signal Aborted (6)

  [1]: https://s3.amazonaws.com/clickhouse-test-reports/33057/19216f4c0ae0f72108c147f958a708b521ad27dc/fuzzer_astfuzzerdebug,actions//report.html

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
test: do not run 'info locals' since 'backtrace full' includes it

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
test: try capture backtrace from all threads

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat commented on docker/test/fuzzer/run-fuzzer.sh:
```diff
@@ -399,3 +398,10 @@ EOF
+for core in core.*; do
+    pigz $core
+    mv $core.gz /output/
+done
```
By the way, I'm not sure that artifacts via `/output` works for `fuzzer`, is this correct?

tavplubix replied:
It's not: https://github.com/ClickHouse/ClickHouse/blob/73a593b67625aa9b478594f2f5d675e215347e25/tests/ci/ast_fuzzer_check.py#L95-L108

azat replied:
Added uploading `core.gz`, and here is a worked example - https://s3.amazonaws.com/clickhouse-test-reports/33057/c43709a12fcca4452331464d31ace4df804e8796/fuzzer_astfuzzerdebug,actions//report.html

azat committed:
test/stress: fix path for core artifacts

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>
Fixes: #33389

azat committed:
test/fuzzer: store core file in artifacts

v2: fix report, because of undefined variable CORE_LINK
v3: fix case when there is no core file

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat commented:
There was one issue with the report for fuzzer, now everything should be correct.
